Team417_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team417_CENTERSTAGE/BaseAutonomous.java

```diff
@@ -0,0 +1,234 @@
+    public static double Y_CALIBRATION_LEFT = 1.0;
+
+
+    public void driveInches(double x, double y) {
+        double xTicks = x * TICKS_PER_INCH * STRAFE_FACTOR;
+        double yTicks = y * TICKS_PER_INCH * DISTANCE_FACTOR;
+
+        double targetFL = xTicks + yTicks;
+        double targetFR = yTicks - xTicks;
+        double targetBL = yTicks - xTicks;
+        double targetBR = yTicks + xTicks;
+
+        // Determine new target position, and pass to motor controller
+        targetFL += FL.getCurrentPosition();
+        targetFR += FR.getCurrentPosition();
+        targetBL += BL.getCurrentPosition();
+        targetBR += BR.getCurrentPosition();
+
+        FL.setTargetPosition((int) targetFL);
+        FR.setTargetPosition((int) targetFR);
+        BL.setTargetPosition((int) targetBL);
+        BR.setTargetPosition((int) targetBR);
+
+        // Turn On RUN_TO_POSITION
+        FL.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+        FR.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+        BL.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+        BR.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+
+        // reset the timeout time and start motion.
+        runtime.reset();
+        FL.setPower(ROBOT_SPEED);
+        FR.setPower(ROBOT_SPEED);
+        BL.setPower(ROBOT_SPEED);
+        BR.setPower(ROBOT_SPEED);
+
+        // keep looping while we are still active, and there is time left, and both motors are running.
+        // Note: We use (isBusy() && isBusy()) in the loop test, which means that when EITHER motor hits
+        // its target position, the motion will stop.  This is "safer" in the event that the robot will
+        // always end the motion as soon as possible.
+        // However, if you require that BOTH motors have finished their moves before the robot continues
+        // onto the next step, use (isBusy() || isBusy()) in the loop test.
+        while (opModeIsActive() &&
+                (runtime.seconds() < 30) &&
+                (FL.isBusy() && FR.isBusy() && BL.isBusy() && BR.isBusy())) {
+        }
+
+        // Stop all motion;
+        stopDriving();
+
+        // Turn off RUN_TO_POSITION
+        FL.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        FR.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        BL.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        BR.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+    }
+
+    OpenCvColorDetection myColorDetection = new OpenCvColorDetection(this);
+
+    public void initializeAuto() {
+        telemetry.addData("Init State", "Init Started");
+        telemetry.update();
+        initializeHardware();
+        myColorDetection.init();
+
+        telemetry.addData("Init State", "Init Finished");
+
+        // Set last know encoder values
+        lastEncoderFR = FR.getCurrentPosition();
+        lastEncoderFL = FL.getCurrentPosition();
+        lastEncoderBL = BL.getCurrentPosition();
+        lastEncoderBR = BR.getCurrentPosition();
+
+        sleep(5000);
```

Comment:
A random sleep without a comment is very sketchy.

Reply:
We added a comment to specify what the sleep does :)